### PR TITLE
New README

### LINKED ISSUE #990: Update README

### PR DESCRIPTION
**Pull request by rasbt**
Here's a draft for the new README. It's based on some great ideas from Adrian's initial draft, like showing only the 3 most recent models. I updated the docs a bit to keep the README more minimal. 

I think the challenge of the old README was that it was too long to read, and most people probably found it overwhelming. There was also lots of duplicated information between the README and the tutorials, and it might be better to not try to show everything in the README but rather show one or two examples and then reference the tutorials.

What are your thoughts?

Fixes #990

### CODE REVIEW

File: tutorials/finetune.md
Comment (carmocca) [inline tutorials/finetune.md:50]:
```suggestion
```

File: tutorials/models.md
Comment (carmocca) [inline tutorials/models.md:1]: Is this file intended? You also added this info to `download_model_weights.md`

File: README.md
Comment (carmocca) [inline README.md:54]: It might look better to center this table like the title is

File: README.md
Comment (carmocca) [inline README.md:28]: Why mention v1?

```suggestion
✅ &nbsp;Optimized and efficient code: Flash Attention v2, multi-GPU support via fully-sharded data parallelism, [optional CPU offloading](tutorials/oom.md#do-sharding-across-multiple-gpus), and [TPU and XLA support](./xla).
```

File: README.md
Comment (carmocca) [inline README.md:75]: zsh friendly

```suggestion
pip install 'litgpt[all]'
```

File: README.md
Comment (carmocca) [inline README.md:81]:
```suggestion
git clone https://github.com/Lightning-AI/litgpt
```

File: README.md
Comment (carmocca) [inline README.md:83]: Lets suggest editable locally

```suggestion
pip install -e '.[all]'
```

File: README.md
Comment (carmocca) [inline README.md:96]:
```suggestion
litgpt download --repo_id mistralai/Mistral-7B-v0.1
```

File: README.md
Comment (carmocca) [inline README.md:99]:
```suggestion
litgpt finetune lora \
```

File: README.md
Comment (carmocca) [inline README.md:108]:
```suggestion
litgpt chat \
```

File: README.md
Comment (carmocca) [inline README.md:127]: I think this only works with raw files. Also, let's link to main

```suggestion
  --config https://raw.githubusercontent.com/Lightning-AI/litgpt/main/config_hub/finetune/llama-2-7b/lora.yaml
```

File: README.md
Comment (carmocca) [inline README.md:126]:
```suggestion
litgpt finetune lora \
```

File: README.md
Comment (carmocca) [inline README.md:134]:
```suggestion
litgpt finetune lora 
```

File: README.md
Comment (carmocca) [inline README.md:135]:
```suggestion
  --config https://raw.githubusercontent.com/Lightning-AI/litgpt/main/config_hub/finetune/llama-2-7b/lora.yaml \
```

File: README.md
Comment (carmocca) [inline README.md:144]: The newline doesnt render

```suggestion
> **Run large models on smaller consumer devices:**
```

File: README.md
Comment (carmocca) [inline README.md:154]:
```suggestion
LitGPT supports rich and customizable [config files](config_hub) to tailor the LLM training to your dataset and hardware needs. Shown below is a configuration file for LoRA finetuning:
```

File: README.md
Comment (carmocca) [inline README.md:315]: Can we make the `tutorials/` directly more prominent? Also include the word "docs" for those who cmd+F it

File: tutorials/models.md
Comment (rasbt) [reply tutorials/models.md:1]: This was accidentally included in the commit

File: tutorials/finetune.md
Comment (awaelchli) [inline tutorials/finetune.md:21]: Should maybe lora move up? Right now it looks like it is the least important.
Full finetuning is not mentioned

File: README.md
Comment (awaelchli) [inline README.md:108]:
```suggestion
litgpt chat \
```

File: README.md
Comment (awaelchli) [inline README.md:109]: This checkpoint dir will depend on whether the user chose to run the optional finetuning step above. 
Also, Mistral-7B-v0.1 is not a chat model so I don't expect it to behave well (or does it?). 

For these commands, I think we should decide if an end-to-end example is more important, or whether individual steps like above are better.

**Review by awaelchli — APPROVED** [review]

File: README.md
Comment (rasbt) [reply README.md:54]: Fixed

File: README.md
Comment (rasbt) [reply README.md:81]: Oh ... not sure what happened there.

File: README.md
Comment (rasbt) [reply README.md:109]: Good point. Let's just use chat here and show finetuning further below.

**Comment by rasbt** [comment]
I believe all your points are addressed now.  Many thanks for the feedback! 

Regarding the LoRA example, @awaelchli , this would be the intended use after your auto-merging PR, correct?

```bash
# 1) Download a pretrained model
litgpt download --repo_id microsoft/phi-2

# 2) Finetune the model
litgpt finetune lora \
  --data Alpaca2k \
  --out_dir out/phi-2-lora \
  --checkpoint_dir checkpoints/microsoft/phi-2

# 3) Chat with the model
litgpt chat \
  --checkpoint_dir out/phi-2-lora/final
```

**Comment by awaelchli** [comment]
@rasbt Yes that would work 🎉

**Pull request merged**
